Cargo.lock

```diff
@@ -220,7 +220,7 @@ dependencies = [
  "arrow-cast",
  "arrow-data",
  "arrow-schema",
- "flatbuffers 22.9.29",
+ "flatbuffers",
```

Comment:
The problem was that we directly depend on `flatbuffers` and also indirectly via `arrow`. Because we were specifying our dependency as `flatbuffers = "*"` - we were pulling latest version v23 while arrow still depends on v22.

I changed our dependency to `flatbuffers = "22"` in order to avoid compiling both versions and use same version as arrow.

There are cargo plugins that allow you to whitelist which crates can be duplicated and which can't - we probably should start using them to ensure we don't compile with duplicated versions of some major crates.